Dockerfile

```
@@ -1,40 +1,25 @@
-# Use OpenJDK 17 as the base image
-FROM openjdk:18-jdk-slim AS builder
+# Use Eclipse Temurin with Maven as the builder image
+FROM maven:3-eclipse-temurin-17-alpine AS builder
```

```suggestion
FROM maven:3-eclipse-temurin-21-alpine AS builder
```
[Commit c255ffc](https://github.com/SkyBlock-Nerds/NerdBot/commit/c255ffc8886703a3f443fb09e48f31cbd1f14d32) broke this

this is on generator tho so doesn't need to be merged but either way I feel like java version should stay the same on every branch aka please change it to 21 on master.

---

also please look into these pr's i don't want to keep on maintaining this, its exhausting.

---

also #387 needs #382 and #375  to also get another update. its just not sustainable to keep on having to go back to the pr's to add comments like these...

so for the love of god look into the pr's